client/telemetry/src/lib.rs

```diff
@@ -245,22 +237,30 @@ impl TelemetryWorker {
 					});
 
 					node.connection_messages.extend(connection_message.clone());
+
+					let (matching, rest): (Vec<_>, Vec<_>) =
+						mem::take(pending_connection_notifications)
```

Comment:
Why not drain?

Reply:
I thought about drain but the drain_filter is only in nightly so I would have to use the `drain(range)` instead. Do you have something in mind shorter and more clever/efficient than that? I would be very curious to see

Reply:
Why do you need to use `drain_filter`? You only need `drain`

Reply:
As I said... drain takes a range. I don't see how you can achieve the same thing with drain... please give me more information

Reply:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=cb8723ec0017bdaa43b192100f776b44

Reply:
That's really not what I am doing here.

Reply:
Fixed in 8aebf81ad